external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -438,55 +440,53 @@ private void commitOffsetsForAckedTuples() {
     // ======== Ack =======
     @Override
     public void ack(Object messageId) {
-        if (!isAtLeastOnce()) {
-            // Only need to keep track of acked tuples if commits are done based on acks
-            return;
-        }
-
+        // Only need to keep track of acked tuples if commits to Kafka are done after a tuple ack is received
```

Comment:
"after a tuple ack is received" -> "based on received tuple acks"

Reply:
Done.